docs/ERC20.md

```diff
@@ -0,0 +1,294 @@
+# ERC20
+
+The ERC20 token standard is a specification for [fungible tokens](https://docs.openzeppelin.com/contracts/4.x/tokens#different-kinds-of-tokens), a type of token where all the units are exactly equal between them. The `ERC20.cairo` contract implements an approximation of [EIP-20](https://eips.ethereum.org/EIPS/eip-20) in Cairo for StarkNet.
+
+## Table of Contents
+
+- [Interface](#interface)
+  * [ERC20 compatibility](#erc20-compatibility)
+- [Usage](#usage)
+- [Extensibility](#extensibility)
+- [API Specification](#api-specification)
+  * [`name`](#-name-)
+  * [`symbol`](#-symbol-)
+  * [`decimals`](#-decimals-)
+  * [`total_supply`](#-total-supply-)
+  * [`balance_of`](#-balance-of-)
+  * [`allowance`](#-allowance-)
+  * [`transfer`](#-transfer-)
+  * [`transfer_from`](#-transfer-from-)
+  * [`approve`](#-approve-)
+
+## Interface
+
+```jsx
+@contract_interface
+namespace IERC20:
+		func name() -> (res: felt):
+    end
+
+    func symbol() -> (res: felt):
+    end
+
+    func decimals() -> (res: felt):
+    end
+
+    func total_supply() -> (res: Uint256):
+    end
+
+    func balance_of(account: felt) -> (res: Uint256):
+    end
+
+    func allowance(owner: felt, spender: felt) -> (res: Uint256):
+    end
+
+    func transfer(recipient: felt, amount: Uint256) -> (success: felt):
+    end
+
+    func transfer_from(
+            sender: felt, 
+            recipient: felt, 
+            amount: Uint256
+        ) -> (success: felt):
+    end
+
+    func approve(spender: felt, amount: Uint256) -> (success: felt):
+    end
+end
+```
+
+### ERC20 compatibility
+
+Although StarkNet is not EVM compatible, this implementation aims to be as close as possible to the ERC20 standard. This is why:
+
+- it uses Cairo's `uint256` instead of `felt`
+- it returns `1` as success to imitate a `bool`
+- it makes use of Cairo's short strings to simulate `name` and `symbol`
+- it will emit events once they're implemented on StarkNet.
+
+Notice that `get_decimals` returns a 252-bit `felt`, meaning it can be much larger than the standard's 8-bit `uint8` and this could potentially affect bridging tokens.
```

Comment:
This is what I think the policy should be:

```suggestion
Notice that `decimals` returns a 252-bit `felt`, meaning it can be much larger than the standard's 8-bit `uint8`. However, compliant implementations should not return a value that is not representable in `uint8`.
```

Are there other differences with ERC20? At least from this document it looks like we're still using snake case? I would consider adding a list of how this implementation _differs_ from ERC20, to complement the above list about how it's similar.